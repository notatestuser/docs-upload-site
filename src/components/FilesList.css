/* tile uploaded pictures */
.upload-list-inline .ant-upload-list-item {
  float: left;
  width: fit-content;
  margin-right: 8px;
}

.upload-list-inline .ant-upload-animate-enter {
  animation-name: uploadAnimateInlineIn;
}

.upload-list-inline .ant-upload-animate-leave {
  animation-name: uploadAnimateInlineOut;
}

@media (max-width: 767px) {
  .upload-list-inline .ant-upload-list-item {
    width: 100% !important;
  }
}
