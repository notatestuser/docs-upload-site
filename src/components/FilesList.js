import React, { Component, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Alert, Upload, Spin, Button, Input } from 'antd';
import filesize from 'filesize';

import './FilesList.css';

import {
  requestFiles,
} from '../redux/files';

const mapStateToProps = ({ files: { files, query, loading, error } }) => ({
  files, query, loading, error,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  requestFiles,
}, dispatch);

class FilesList extends Component {
  static defaultProps = {
    files: [],
  }

  componentDidMount() {
    const { requestFiles } = this.props;
    requestFiles();
  }

  render() {
    const { files, loading, error } = this.props;
    if (loading) {
      return <Spin size="large" />;
    }
    if (error) {
      return (
        <Alert
          message="Error"
          description="An error has occurred. Is the server reachable?"
          type="error"
          showIcon
        />
      );
    }
    if (!files.length) {
      return (
        <Alert
          message="No files available"
          description="There are no files uploaded."
          type="info"
          showIcon
        />
      );
    }
    const fileList = files.map(({ name, size }, idx) => ({
      uid: idx,
      name: `${name} (${filesize(size)})`,
      status: 'done',
      url: `/uploads/${name}`,
    }));
    const props = {
      action: '',
      listType: 'picture',
      defaultFileList: [...fileList],
      className: 'upload-list-inline',
    };
    return (
      <Fragment>
        <Input.Search
          placeholder="input and press enterâ€¦"
          defaultValue={this.props.query}
          onSearch={value => {
            this.props.requestFiles(value);
          }}
          style={{ width: 200 }}
        />
        {this.props.query ? (
          <Button
            type="primary"
            onClick={() => { this.props.requestFiles(); }}
          >
            Clear
          </Button>
        ) : null}
        <Upload {...props} />
      </Fragment>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilesList);
