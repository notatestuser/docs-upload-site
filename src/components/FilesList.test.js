import React from 'react';
import renderer from 'react-test-renderer';
import { FilesList } from './FilesList';

it('passes a snapshot test with 2 files', () => {
  const props = {
    files: [
      { name: "file1.jpg", size: 1 },
      { name: "file2.pdf", size: 2 },
    ],
    loading: false,
    requestFiles: () => {},
  }
  const tree = renderer.create(
    <FilesList {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders list items for 2 files', () => {
  const props = {
    files: [
      { name: "file1.jpg", size: 1 },
      { name: "file2.pdf", size: 2 },
    ],
    loading: false,
    requestFiles: () => {},
  }
  const rendered = renderer.create(
    <FilesList {...props} />
  );
  const instance = rendered.root;
  expect(instance.findByProps({ title: 'file1.jpg (1 B)' }).type).toEqual('a');
  expect(instance.findByProps({ title: 'file2.pdf (2 B)' }).type).toEqual('a');
});
