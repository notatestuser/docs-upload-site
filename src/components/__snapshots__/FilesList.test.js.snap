// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes a snapshot test with 2 files 1`] = `
<span
  className="upload-list-inline"
>
  <div
    className="ant-upload ant-upload-select ant-upload-select-picture"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <span
      className="ant-upload"
      onClick={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={undefined}
      tabIndex="0"
    >
      <input
        accept=""
        directory={null}
        multiple={false}
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
        webkitdirectory={null}
      />
    </span>
  </div>
  <div
    className="ant-upload-list ant-upload-list-picture"
    style={undefined}
  >
    <div
      className="ant-upload-list-item ant-upload-list-item-done"
    >
      <div
        className="ant-upload-list-item-info"
      >
        <span>
          <a
            className="ant-upload-list-item-thumbnail"
            href="/uploads/file1.jpg"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="file1.jpg (1 B)"
              src="/uploads/file1.jpg"
            />
          </a>
          <a
            className="ant-upload-list-item-name"
            href="/uploads/file1.jpg"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="file1.jpg (1 B)"
          >
            file1.jpg (1 B)
          </a>
        </span>
      </div>
      <i
        className="anticon anticon-cross"
        onClick={[Function]}
        title="Remove file"
      />
    </div>
    <div
      className="ant-upload-list-item ant-upload-list-item-done"
    >
      <div
        className="ant-upload-list-item-info"
      >
        <span>
          <a
            className="ant-upload-list-item-thumbnail"
            href="/uploads/file2.pdf"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              className="anticon anticon-file ant-upload-list-item-icon"
            />
          </a>
          <a
            className="ant-upload-list-item-name"
            href="/uploads/file2.pdf"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="file2.pdf (2 B)"
          >
            file2.pdf (2 B)
          </a>
        </span>
      </div>
      <i
        className="anticon anticon-cross"
        onClick={[Function]}
        title="Remove file"
      />
    </div>
  </div>
</span>
`;
